lib/http/request.js

```diff
@@ -262,6 +262,16 @@ class HttpRequest extends EventEmitter {
       return;
     }
 
+    if (this.httpResponse.isInternalServerError && this.retryAttempts > 0) {
+      this.retryCount = this.retryCount + 1;
+      setTimeout(()=> {
+        this.retryAttempts = this.retryAttempts - 1;
+        this.send();
+      }, 1000);
```

Comment:
this should be configurable so we don't add so much delay in the tests.

Reply:
`retryRequestInterval` as a configuration parameter